packages/suite-desktop-core/e2e/tests/onboarding/t2t1-recovery-success.test.ts

```diff
@@ -0,0 +1,58 @@
+import { test, expect } from '../../support/fixtures';
+
+test.describe('Onboarding - recover wallet T2T1', { tag: ['@group=device-management'] }, () => {
+    test.use({
+        emulatorStartConf: { wipe: true, model: 'T2T1', version: '2.5.3' },
+        emulatorSetupConf: undefined,
+    });
+    test.beforeEach(async ({ analyticsPage, onboardingPage }) => {
+        await onboardingPage.disableFirmwareHashCheck();
+
+        // Go through analytics opt-out
+        analyticsPage.continue();
+        analyticsPage.continue();
+    });
+
+    test('Successfully recovers wallet from mnemonic', async ({
+        onboardingPage,
+        page,
+        devicePrompt,
+        trezorUserEnvLink,
+    }) => {
+        // Start wallet recovery process and confirm on device
+        await onboardingPage.skipFirmware();
+        await onboardingPage.recoverWalletButton.click();
+        await onboardingPage.startRecoveryButton.click();
+        await devicePrompt.confirmOnDevicePromptIsShown();
+
+        await page.waitForTimeout(1000);
+        await trezorUserEnvLink.pressYes();
+
+        await devicePrompt.confirmOnDevicePromptIsShown();
+        await page.waitForTimeout(1000);
+        await trezorUserEnvLink.pressYes();
+
+        await page.waitForTimeout(1000);
+        await trezorUserEnvLink.selectNumOfWordsEmu(12);
+
+        await page.waitForTimeout(1000);
+        await trezorUserEnvLink.pressYes();
+
+        // Input mnemonic
+        await page.waitForTimeout(1000);
+        for (let i = 0; i < 12; i++) {
+            await trezorUserEnvLink.inputEmu('all');
+        }
+
+        // Confirm recovery success
+        await page.waitForTimeout(1000);
+        await trezorUserEnvLink.pressYes();
+
+        // Finalize recovery, skip pin, and check success
+        await onboardingPage.continueRecoveryButton.click();
+        await onboardingPage.skipPin();
+        await onboardingPage.continueCoinsButton.click();
+        await expect(onboardingPage.finalTitle).toBeVisible();
+        await expect(onboardingPage.finalTitle).toContainText('Setup complete!');
```

Comment:
I bet toContainText checks for visibility too. so we can remove the expect visible
And if we have exact string then I would go with toHaveText as that will eliminate the chance of false pass with partial match of toContainText

Reply:
I think it doesn't check for visibility and the string is not full afaik.

Reply:
This is a shock for me. I will do later some investigation.
So far I did some tests and seems that toContainText will pass even on hidden element
but toHaveText would fail.
wow.